autoscale/src/main/java/com/sequenceiq/periscope/monitor/CleanupMonitor.java

```diff
@@ -0,0 +1,48 @@
+package com.sequenceiq.periscope.monitor;
+
+import static com.sequenceiq.periscope.api.model.ActivityStatus.SCALING_FLOW_FAILED;
+import static com.sequenceiq.periscope.api.model.ActivityStatus.SCALING_FLOW_SUCCESS;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.periscope.api.model.ActivityStatus;
+import com.sequenceiq.periscope.model.ScalingActivities;
+import com.sequenceiq.periscope.monitor.evaluator.EvaluatorExecutor;
+import com.sequenceiq.periscope.monitor.evaluator.cleanup.ActivityCleanupEvaluator;
+
+@Component
+@ConditionalOnProperty(prefix = "periscope.enabledAutoscaleMonitors.cleanup-monitor", name = "enabled", havingValue = "true")
+public class CleanupMonitor extends ScalingActivityMonitor {
```

Comment:
One overall question here. When there's multiple Periscope instances - will this run in all the instances?, or just on one of them.
Will there be conflicts / failures if running on multiple instances ?

Reply:
Updated this as discussed. With the new patch, only the periscope node elected as the leader will perform cleanup. Reproduced the HA setup by running multiple periscope instances locally via IDEA. Additionally, added a null check to ensure cleanup works for non HA local setups